sound/soc/sof/intel/hda-dai.c

```diff
@@ -419,11 +440,19 @@ static int ssp_dai_hw_params(struct snd_pcm_substream *substream,
 
 		if (!strcmp(dai->name, sof_dai->cpu_dai_name) &&
 		    substream->stream == sof_dai->comp_dai.direction) {
-			config = &sof_dai->dai_config[sof_dai->current_config];
-			if (!config) {
-				dev_err(sdev->dev, "error: no config for DAI %s\n", sof_dai->name);
+			if (sof_dai->current_config > sof_dai->number_configs - 1) {
+				dev_err(sdev->dev, "error: DAI config index out of range\n");
 				return -EINVAL;
 			}
+			/* we can return immediately as there are no choises */
+			if (sof_dai->number_configs < 2)
+				config = &sof_dai->dai_config[sof_dai->current_config];
+			/* try to match to pcm params */
+			if (!config)
+				config = ssp_dai_config_pcm_params_match(sdev, sof_dai, params);
+			/* last resort set default config */
+			if (!config)
+				config = &sof_dai->dai_config[sof_dai->current_config];
```

Comment:
why not flip the logic so you do the params matching first. If no match found, set default config. You do the same thing is the first and the third if's anyway

Reply:
this will now change